src/Test/Utilities/Desktop/CommonDiagnosticAnalyzers.cs

```diff
@@ -91,67 +81,69 @@ public static string GetExpectedErrorLogIssuesText(Compilation compilation)
+            GetExpectedPropertiesMapText() + @"
+          }
+        },
+        {
+          ""ruleId"": """ + Descriptor2.Id + @""",
+          ""locations"": [
+          ],
+          ""shortMessage"": """ + Descriptor2.MessageFormat + @""",
+          ""fullMessage"": """ + Descriptor2.Description + @""",
+          ""properties"": {
+            ""severity"": """ + Descriptor2.DefaultSeverity + @""",
+            ""defaultSeverity"": """ + Descriptor2.DefaultSeverity + @""",
+            ""title"": """ + Descriptor2.Title + @""",
+            ""category"": """ + Descriptor2.Category + @""",
+            ""helpLink"": """ + Descriptor2.HelpLinkUri + @""",
+            ""isEnabledByDefault"": """ + Descriptor2.IsEnabledByDefault + @""",
+            ""isSuppressedInSource"": ""False"",
+            ""customTags"": """ + Descriptor2.CustomTags.Join(";") + @"""" +
+            GetExpectedPropertiesMapText() + @"
+          }
         }
-      ],
-      ""shortMessage"": """ + Descriptor1.MessageFormat + @""",
-      ""fullMessage"": """ + Descriptor1.Description + @""",
-      ""properties"": {
-        ""severity"": """ + Descriptor1.DefaultSeverity + @""",
-        ""warningLevel"": ""1"",
-        ""defaultSeverity"": """ + Descriptor1.DefaultSeverity + @""",
-        ""title"": """ + Descriptor1.Title + @""",
-        ""category"": """ + Descriptor1.Category + @""",
-        ""helpLink"": """ + Descriptor1.HelpLinkUri + @""",
-        ""isEnabledByDefault"": """ + Descriptor1.IsEnabledByDefault + @""",
-        ""isSuppressedInSource"": ""False"",
-        ""customTags"": """ + Descriptor1.CustomTags.Join(";") + @"""" +
-        GetExpectedPropertiesMapText() + @"
-      }
-    },
-    {
-      ""ruleId"": """ + Descriptor2.Id + @""",
-      ""locations"": [
-      ],
-      ""shortMessage"": """ + Descriptor2.MessageFormat + @""",
-      ""fullMessage"": """ + Descriptor2.Description + @""",
-      ""properties"": {
-        ""severity"": """ + Descriptor2.DefaultSeverity + @""",
-        ""defaultSeverity"": """ + Descriptor2.DefaultSeverity + @""",
-        ""title"": """ + Descriptor2.Title + @""",
-        ""category"": """ + Descriptor2.Category + @""",
-        ""helpLink"": """ + Descriptor2.HelpLinkUri + @""",
-        ""isEnabledByDefault"": """ + Descriptor2.IsEnabledByDefault + @""",
-        ""isSuppressedInSource"": ""False"",
-        ""customTags"": """ + Descriptor2.CustomTags.Join(";") + @"""" +
-        GetExpectedPropertiesMapText() + @"
-      }
+      ]
     }
   ]
 }";
             }
 
-            public static string EscapeDirectorySeparatorChar(string input)
+            public static string GetEscapedUriForPath(string path)
             {
-                return input.Replace(Path.DirectorySeparatorChar.ToString(), @"\" + Path.DirectorySeparatorChar);
+                return new Uri(path, UriKind.RelativeOrAbsolute).ToString().Replace("/", "\\/");
```

Comment:
Aesthetically, I'd rather we didn't, but it's done by the DataContractJsonSerializer and I can't find a way to override it.